tests/edit.bats

```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test flox edit
+# TODO move other edit tests from integration.bats
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash;
+
+# bats file_tags=edit
+
+# ---------------------------------------------------------------------------- #
+
+EDIT_ENVIRONMENT=_edit_testing_
+
+# ---------------------------------------------------------------------------- #
+
+@test "flox edit -f" {
+  run $FLOX_CLI destroy --force -e "$EDIT_ENVIRONMENT"
+  run $FLOX_CLI create -e "$EDIT_ENVIRONMENT"
+  assert_success
+
+  # test reading from a file
+  run $FLOX_CLI edit -e "$EDIT_ENVIRONMENT" -f "$TESTS_DIR/test-flox.nix"
+  assert_success
+  assert_output --partial "Environment '$EDIT_ENVIRONMENT' modified."
+
+  EDITOR=cat run $FLOX_CLI edit -e "$EDIT_ENVIRONMENT"
+  assert_success
+  assert_output --partial 'environmentVariables.test = "file"'
+
+  # test reading from stdin
+  contentsStdin=$(cat <<'EOF'
+{
+  environmentVariables.test = "stdin";
+}
+EOF
+)
+  run $FLOX_CLI edit -e "$EDIT_ENVIRONMENT" --file - <<<"$contentsStdin"
+  assert_success
+  assert_output --partial "Environment '$EDIT_ENVIRONMENT' modified."
+  EDITOR=cat run $FLOX_CLI edit -e "$EDIT_ENVIRONMENT"
```

Comment:
```suggestion
  run sh -c "EDITOR=cat $FLOX_CLI edit -e '$EDIT_ENVIRONMENT';";
```

Reply:
What's the advantage of doing it that way?

Reply:
It's a wonky `bats` quirk.
`run` is a wrapper around `eval` that doesn't pass through env variables in a consistent manner when different versions of `sh` are used.
We conventionally use `bash@5.x` on x86_64-linux, but most Darwin machines are going to have `bash@3.x`, and Ubuntu/WSL boxes are going to have `bash@4.x`.

In short "I'm recommending the change because it's mentioned in the FAQ/quirks section of the `bats` manual as a potential pitfall". It likely works fine but if the authors went out of their way to make a recommendation I'm happy to follow it.

Reply:
Why are we not running these with nix supplied bash?
Its exactly those OS wonkynesses where nix is supposed to help with consistency.
If these tests run with your system bash were doing something wrong from the start IMO